symengine/pow.h

```diff
@@ -34,12 +34,12 @@ class Pow : public Basic
     //! \return `true` if canonical
     bool is_canonical(const Basic &base, const Basic &exp) const;
     //! \return `base` of `base**exp`
-    inline RCP<const Basic> get_base() const
+    inline const RCP<const Basic>& get_base() const
     {
         return base_;
     }
     //! \return `exp` of `base**exp`
-    inline RCP<const Basic> get_exp() const
+    inline const RCP<const Basic>& get_exp() const
```

Comment:
I think a safe solution might be to use Viewable (https://github.com/certik/trilinos/pull/2), I need to finish that PR. 

P.S. you can't do `RCP<const Basic> a = get_exp(); ` if `get_exp()` only returns a reference `Basic &`.